client/hawc_client/__init__.py

```diff
@@ -255,6 +285,48 @@ def references(self, assessment_id: int) -> pd.DataFrame:
         response_json = self.session.get(url).json()
         return pd.DataFrame(response_json)
 
+    def update_reference(self, reference_id: int, **kwargs) -> Dict:
+        """
+        Updates reference with given values. Fields not passed as parameters
+        are unchanged.
+
+        Args:
+            reference_id (int): ID of reference to update
+            **kwargs (optional): Named parameters of fields to update in reference. Example parameters:
+                title (str): title of the reference
+                abstract (str): reference abstract
+                tags (List[int]): tag IDs to apply to reference;
+                    completely replaces current tags
+
+        Example Usage:
+            updated_reference_json = client.lit.update_reference(
+                reference_id = 1,
+                title = "reference",
+                tags = [1,2,3]
+            )
+
+        Returns:
+            Dict: JSON representation of the updated reference.
+        """
+        url = f"{self.session.root_url}/lit/api/reference/{reference_id}/"
+        response_json = self.session.patch(url, kwargs).json()
+        return response_json
+
+    def delete_reference(self, reference_id: int) -> None:
+        """
+        Deletes the given reference. This also removes the reference from
```

Comment:
suggested revision; `Deletes the selected reference. This also removes the reference from any searches/imports which may have included the reference. If data was extracted with this reference and it is associated with bioassay, or epi extractions they will also be removed.`